src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -500,7 +518,8 @@ private bool ReplacementChangesSemanticsForNode(TSyntaxNode currentOriginalNode,
             {
                 var originalExpression = (TExpressionSyntax)currentOriginalNode;
                 var newExpression = (TExpressionSyntax)currentReplacedNode;
-                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression))
+                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression) ||
+                    ReplacementIntroducesErrorType(originalExpression, newExpression))
                 {
                     return true;
```

Comment:
ignore my comment, it was due to git diff, code actually belong to different method.